service/history/queues/speculative_workflow_task_timeout_queue.go

```diff
@@ -94,19 +95,9 @@ func (q SpeculativeWorkflowTaskTimeoutQueue) Category() tasks.Category {
 func (q SpeculativeWorkflowTaskTimeoutQueue) NotifyNewTasks(ts []tasks.Task) {
 	for _, task := range ts {
 		if wttt, ok := task.(*tasks.WorkflowTaskTimeoutTask); ok {
-			executable := newSpeculativeWorkflowTaskTimeoutExecutable(NewExecutable(
-				0,
-				wttt,
-				q.executor,
-				nil,
-				nil,
-				q.priorityAssigner,
-				q.timeSource,
-				q.namespaceRegistry,
-				q.clusterMetadata,
-				q.logger,
-				q.metricsHandler,
-			), wttt)
+			executable := newSpeculativeWorkflowTaskTimeoutExecutable(
+				q.executableFactory.NewExecutable(wttt, 0), wttt,
```

Comment:
we probably don't want to DLQ an in-memory timer task, at least for now. I actually haven't thought about this case.

Reply:
Sounds good. Removed.